test/core/bad_client/bad_client.cc

```diff
@@ -57,7 +57,7 @@ static void thd_func(void* arg) {
 }
 
 /* Sets the done_write event */
-static void set_done_write(void* arg, grpc_error* error) {
+static void set_done_write(void* arg, grpc_error* /*error*/) {
```

Comment:
It's invoked through a grpc_closure and all closures have to have the same arguments so that they can be dispatched (and that's a generic argument and an error). At least until Yash's closure changes come down the pipe.